src/components/OSCALControlProse.js

```diff
@@ -259,27 +272,159 @@ export function OSCALReplacedProseWithParameterLabel(props) {
+    RegExp(prosePlaceholderRegexpString, "g")
+  )[1];
+
   return (
     <Typography className={props.className}>
-      {props.label}
-      {props.prose
+      {labelWithProse
         .split(RegExp(prosePlaceholderRegexpString, "g"))
         .map((segment, index) => {
           if (index % 2 === 0) {
             return getTextSegment(segment, index.toString());
           }
-          return getParameterLabelSegment(
-            props.parameters,
-            segment,
-            props.modificationSetParameters,
-            index.toString()
+
+          return (
+            <Typography display="inline" variant="body2">
+              {getParameterLabelSegment(
+                props.parameters,
+                segment,
+                props.modificationSetParameters,
+                index.toString()
+              )}
+            </Typography>
           );
         })}
       {props.modificationDisplay}
+      {props.label && props.isEditable ? (
+        <IconButton
+          onClick={(event) => {
+            setAnchorEl(event.currentTarget);
+          }}
+        >
+          <EditIcon fontSize="small" />
+        </IconButton>
+      ) : null}
+      {props.isEditable ? (
+        <Grid container>
+          <OSCALPopover
+            anchorEl={anchorEl}
+            setAnchorEl={setAnchorEl}
+            controlId={props.controlId}
+            onCancel={() => {
+              setAnchorEl(null);
+            }}
+            onFieldSave={(descriptionReference, implementationReference) => {
+              const newByComponent = {
+                "component-uuid": props.componentId,
+                uuid: v4(),
+                description: descriptionReference.current.value,
+              };
+
+              statement["by-components"].push(newByComponent);
+
+              const editedField = [
+                "statements",
+                `statement-id[${props.statementId}]`,
+                "by-components",
+                `component-uuid[${props.componentId}]`,
+              ];
+
+              if (paramId) {
+                newByComponent["set-parameters"] = [
+                  {
+                    "param-id": paramId,
+                    values: [implementationReference.current.value],
+                  },
+                ];
+
+                editedField.push(
+                  "set-parameters",
+                  `param-id[${paramId}]`,
+                  "values"
+                );
+              } else {
+                editedField.push("description");
+              }
+
+              props.onFieldSave(
+                false,
+                partialRestData,
+                props.update,
+                editedField,
+                null,
+                "PUT",
+                `${rootOscalObjectName}/${props.restData[rootOscalObjectName].uuid}/control-implementation/implemented-requirements/${props.implementedRequirement.uuid}`
+              );
+            }}
```

Comment:
We're eventually want to make our own sort of API client library at some point so it's not so hardcoded into the components.